polaris-react/src/components/Navigation/components/Item/Item.tsx

```diff
@@ -84,6 +85,20 @@ export function Item({
     </span>
   ) : null;
 
+  const matchState = matchStateForItem(
+    {url, matches, exactMatch, matchPaths, excludePaths},
+    location,
+  );
+
+  const selected =
+    selectedOverride == null
+      ? matchState === MatchState.MatchForced ||
+        matchState === MatchState.MatchUrl ||
+        matchState === MatchState.MatchPaths
+      : selectedOverride;
+
+  const icon = selected ? matchedItemIcon || baseIcon : baseIcon;
```

Comment:
Deliberately not choosing to gate this behind the feature flag since it's an optional prop that isn't going to impact the current state of the Admin.